packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -874,29 +921,25 @@ export class IndexedDbPersistence implements Persistence {
    * cleanup logic in `shutdown()`.
    */
   private isClientZombied(clientId: ClientId): boolean {
-    if (this.window.localStorage === undefined) {
-      assert(
-        process.env.USE_MOCK_PERSISTENCE === 'YES',
-        'Operating without LocalStorage is only supported with IndexedDbShim.'
-      );
-      return false;
-    }
-
-    try {
-      const isZombied =
-        this.window.localStorage.getItem(
-          this.zombiedClientLocalStorageKey(clientId)
-        ) !== null;
-      log.debug(
-        LOG_TAG,
-        `Client '${clientId}' ${
-          isZombied ? 'is' : 'is not'
-        } zombied in LocalStorage`
-      );
-      return isZombied;
-    } catch (e) {
-      // Gracefully handle if LocalStorage isn't available / working.
-      log.error(LOG_TAG, 'Failed to get zombied client id.', e);
+    if (this.webStorage) {
+      try {
+        const isZombied =
+          this.webStorage.getItem(
+            this.zombiedClientLocalStorageKey(clientId)
+          ) !== null;
+        log.debug(
+          LOG_TAG,
+          `Client '${clientId}' ${
+            isZombied ? 'is' : 'is not'
+          } zombied in LocalStorage`
+        );
+        return isZombied;
+      } catch (e) {
+        // Gracefully handle if LocalStorage isn't working.
+        log.error(LOG_TAG, 'Failed to get zombied client id.', e);
+        return false;
+      }
+    } else {
```

Comment:
tl;dr: return early for trivial cases.

Alternatively, you can check the negation of this if-condition and move the return early and move the try-statement out of the if. Does not matter for the code itself but will help human understanding the code.

Reply:
This has also been refactored away.